proposals/lip-0069.md

```diff
@@ -61,11 +61,11 @@ _Endpoint_ is the interface for a module to an external system through an RPC en
 
 #### Application hooks
 
-Application hooks are module methods which are called during the block execution. These hooks cannot mutate the block, but they can introduce state changes.
+Application hooks are module methods which are called during the block execution. These hooks can access the current state, the block header being currently executed and its block assets, and make state mutations.
```

Comment:
If we say that these hooks access the block assets, shouldn't we also say that they access the block transactions? I think what we want to say is that they can access the current state and the block header to process the state transitions induced by the block assets and the block transactions.

Reply:
So basically it is the whole block and the state. Depending on the scope of the hook, you can only access a part of the block, e.g., one transaction and the block header, or one asset and the block header.

Reply:
Maybe we could write something like the following:

> Application hooks are module methods which are called during the block execution. In general, these hooks can access the current state and the block being currently executed, and make state mutations. In most cases we restrict the access to the state and block to have a clean and modular framework for state mutations.  For instance, in the hook for executing a command only the containing transaction, certain properties in the block header and the state of the respective module is accessible.

Reply:
Can be resolved from my side.